src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
@dol I made required changes to in my composer fork https://github.com/realshadow/composer/commit/5b748eda6cc87501aefeff3692ba9ffc077422dd. Tested it with `VcsRepository`, `PearRepository` and `ComposerRepository`, I kindly expect it to work with `ArtifactRepository` and `PathRepository` as well. Works like a charm. Should I try to make a PR to composer or shall we try to solve this in satis?